src/Symfony/Component/AssetMapper/Compressor/GzipCompressor.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AssetMapper\Compressor;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Process\Process;
+
+/**
+ * Compresses a file using zopfli if possible, or fallback on gzip.
+ *
+ * @author Kévin Dunglas <kevin@dunglas.dev>
+ */
+final class GzipCompressor implements SupportedCompressorInterface
+{
+    use CompressorTrait {
+        compress as baseCompress;
+    }
+
+    private const WRAPPER = 'compress.zlib';
+    private const COMMAND = 'gzip';
+    private const PHP_EXTENSION = 'zlib';
+    private const FILE_EXTENSION = 'gz';
+
+    public function __construct(
+        private readonly ZopfliCompressor $zopfliCompressor = new ZopfliCompressor(),
+        ?string $executable = null,
+        private ?LoggerInterface $logger = null,
+    ) {
+        $this->executable = $executable;
+    }
+
+    public function compress(string $path): void
+    {
+        if (null === $reason = $this->zopfliCompressor->getUnsupportedReason()) {
```

Comment:
shouldn't the `getUnsupportedReason` implementation also check whether zopfli is supported ? Maybe your server has the zopfli executable but does not have the gzip one and the zlib extension. In such case, the current implementation would mark this compressor as unsupported while the `compress` method would work